include/odp/api/spec/classification.h

```diff
@@ -178,6 +179,60 @@ typedef struct odp_bp_param_t {
 	odp_threshold_t threshold;
 } odp_bp_param_t;
 
+/**
+ * Classifier queue specific statistics counters
+ */
+typedef struct odp_cls_queue_stats_t {
+	/** Number of octets in packets delivered to the queue. In case of
+	 *  Ethernet, packet size includes MAC header and FCS. */
+	uint64_t octets;
+
+	/** Number of packets delivered to the queue. */
+	uint64_t packets;
+
+	/** Number of packets delivered to the queue which were discarded due to
+	 *  other reasons (e.g. RED) than errors. */
+	uint64_t discards;
+
+	/** Number of packets delivered to the queue which contained errors. */
+	uint64_t errors;
+
+} odp_cls_queue_stats_t;
+
+/**
+ * Classifier statistics capabilities
+ */
+typedef struct odp_cls_stats_capability_t {
+	/** Queue level capabilities */
+	struct {
+		/** Supported counters */
+		union {
+			/** Statistics counters in a bit field structure */
+			struct {
+				/** @see odp_cls_queue_stats_t::octets */
+				uint64_t octets          : 1;
+
+				/** @see odp_cls_queue_stats_t::packets */
+				uint64_t packets         : 1;
+
+				/** @see odp_cls_queue_stats_t::discards */
+				uint64_t discards        : 1;
+
+				/** @see odp_cls_queue_stats_t::errors */
+				uint64_t errors          : 1;
+
+			} counter;
+
+			/** All bits of the bit field structure
+			 *
+			 *  This field can be used to set/clear all flags, or
+			 *  for bitwise operations over the entire structure. */
+			uint64_t all_counters;
+		};
+	} queue;
+
+} odp_cls_stats_capability_t;
```

Comment:
Are you planning to use other statistics elements in this structure? If not, then, probably, no need for the internal "queue" structure.

Reply:
Yes, additional statistics could be added later (e.g. CoS).